Framework/DataHandling/src/LoadBBY.cpp

```diff
@@ -471,52 +450,105 @@ std::vector<bool> LoadBBY::createRoiVector(const std::string &maskfile) {
   return result;
 }
 
+// loading instrument parameters
+void LoadBBY::loadInstrumentParameters(
+    NeXus::NXEntry &entry, std::map<std::string, double> &logParams,
+    std::map<std::string, std::string> &allParams) {
+
+  std::string idfDirectory =
+      Mantid::Kernel::ConfigService::Instance().getString(
+          "instrumentDefinition.directory");
+
+  try {
+    std::string parameterFilename = idfDirectory + "BILBY_Parameters.xml";
+    // Set up the DOM parser and parse xml file
+    Poco::XML::DOMParser pParser;
```

Comment:
I think you might be able to avoid having to manually parse the xml parameters file here. When the LoadInstrument algorithm is called it automatically calls LoadParameterFile as a child algorithm. This does the parsing of the parameters xml file for you and stores the results in the instrument object which is associated with the workspace. 

LoadInstrument is called at the end of this algorithm currently but I think it should be possible to call it earlier so these parameters are available from eventWS at this stage. 

This is only a suggestion though as the current implementation works it just somewhat duplicates the functionality here https://github.com/mantidproject/mantid/blob/1b42a7e3fc4c9babcc36ad7562e60fed364b2f7b/Framework/DataHandling/src/LoadParameterFile.cpp

Reply:
Hi Matt, the only reason I included the parsing initially was to setup the log file correctly prior to loading the instrument so that the parameter values in the IDF were initialized. The XML file does not contain the values - only the names of the parameters in the HDF file. I am not aware of another way of solving my problem.

Reply:
Hi Geish, good point if you need to set up the logs correctly prior to loading the IDF then there doesn't seem to be an easy way around this.